tests/phpunit/tests/rest-api/test-notification-controller.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Unit tests covering WP_REST_Notification_Controller functionality.
+ */
+class WP_Test_REST_Notification_Controller extends WP_Test_REST_Post_Type_Controller_Testcase {
```

Comment:
Do we want to extend the post type controller test case here? I'd have thought we'd use `WP_Test_REST_Controller` as a starting point, given we haven't used the post controller for the notifications controller itself.